utils/validate_images.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+const core = require('@actions/core');
+const {promises:fs, statSync} = require("fs");
+const path = require('path');
+const isImage = require('is-image');
+
+const BASE_PATH = './packs/';
+const MAX_SIZE = 4000000;
+const MAX_NUM_IMG = 6;
+const ALLOWED_IMG_EXT = [
+  '.png',
+  '.jpeg',
+  '.jpg',
+  '.svg',
+];
+let valid = true,
+    sizeErrors = [],
+    typeErrors = [],
+    tooManyImages = [];
+    
+/** 
+* Validates an object against a JSON schema
+* @param {String} dir - The directory to parse, set by the command line argument
+* @returns {Object[]} An array of any errors found
+*/
+async function getFiles(dir) {
```

Comment:
I'm not 100% sure this will work for you, but `glob` might simplify a lot of this logic. Take a look at the `validate_packs.js` script, there's a bunch of globbing going on there

Reply:
Ye i was trying to use as few packages as possible but I guess it doesn't really matter if it's just for the actions, have integrated Glob now and slimmed it down but let me know if there's other places I can streamline it